cmd/crane/cmd/mutate.go

```diff
@@ -139,3 +155,23 @@ func validateKeyVals(kvPairs map[string]string) error {
 	}
 	return nil
 }
+
+// setEnvVars override envvars in a config
+func setEnvVars(cfg *v1.Config, envVars map[string]string) error {
```

Comment:
This implementation will shuffle the order of environment variables due to golang's random map iteration order. To be safe, I think we want to preserve the existing order.

Also, I believe linux environment variable keys are case sensitive, but windows keys are case insensitive.

Reply:
I can sort alphabetically -in order to keep stable - but I don't think preserving is worth the extra code do you?
Re windows - as  crane and OCI support multi-arch what would you prefer?
I was thinking as anyway this is a util - we can validate that all new ones are upper-case (the convention on Linux and enforced on Windows) - and roll with that - or we only convert/filter if there is a Windows  image.

WYDT?

Reply:
@jonjohnsonjr waiting to see regarding case sensitivity

Reply:
> I can sort alphabetically -in order to keep stable - but I don't think preserving is worth the extra code do you?

I'd like to keep the original order if we can. The approach I'd take is to iterate over the original list, replace any value if it's in the new `envVars` map and remove that from `envVars`. If `envVars` isn't not empty at the end, append any remaining values.

I don't care about the order of _new_ environment variables that much. I mostly want to avoid disturbing the order of existing environment variables.

> or we only convert/filter if there is a Windows image

I think I'd prefer checking for windows (like we do in [append](https://github.com/google/go-containerregistry/blob/c75901cce6272aea6bbcd9b517ba688d63a413f5/pkg/crane/append.go#L45)) and do case-insensitive comparison.

For windows, I'd uppercase all new env keys in the map so we can just do something like:

```
old := split[0]
if windows {
  old = strings.ToUpper(old)
}
if v, ok := envVars[old]; ok {
  split[1] = v 
}
```

Reply:
Done